gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt

```diff
@@ -0,0 +1,34 @@
+// connection initiated by packetdrill
+--tolerance_usecs=200000
+`../common/defaults.sh`
+
++0     `sysctl net.mptcp.add_addr_timeout=1`
++0     `../common/multi-ep.sh -e 1 -m signal`
+
++0     socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0     setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
+
++0     bind(3, ..., ...) = 0
++0     listen(3, 1) = 0
++0       <  S   0:0(0)         win 32792  <mss 1460, sackOK, nop, nop, nop, wscale 7, mpcapable v1 flags[flag_h] nokey>
++0       >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey]>
++0.01    <   .  1:1(0)  ack 1  win 257                                               <mpcapable v1 flags[flag_h] key[ckey=2, skey]>
++0     accept(3, ..., ...) = 4
+
+// ADD_ADDR is sent directly (>= 5.12), not after first data (< 5.12)
++0       >   .  1:1(0)        ack 1              <add_address address_id=1  addr[saddr0] hmac=auto,                 dss dack4=1 ssn=1 dll=0 nocs>
+// send echo with wrong address, expect a retry in 1s
++0       <   .  1:1(0)        ack 1     win 257  <add_address address_id=1 addr[ep=inet_addr("1.2.3.4")] addr_echo, dss dack4=1 ssn=1 dll=0 nocs>
++1.0     >   .  1:1(0)        ack 1              <add_address address_id=1  addr[saddr0] hmac=auto,                 dss dack4=1 ssn=1 dll=0 nocs>
+// send echo with wrong id, expect a retry in 1s
++0       <   .  1:1(0)        ack 1     win 257  <add_address address_id=90 addr[saddr0] addr_echo>
++1.0     >   .  1:1(0)        ack 1              <add_address address_id=1  addr[saddr0] hmac=auto,                 dss dack4=1 ssn=1 dll=0 nocs>
+// send echo with wrong id, expect  no retry
```

Comment:
with "right" ID? (same in the `_v6` version)

Also, do we not retry max 3 times?

Reply:
> with "right" ID? (same in the `_v6` version)

correct,

> Also, do we not retry max 3 times?

we do, but AFAIR the fisrt attempt is not accounted as a retry. Not 100% sure though, I'll try locally echoing address_id=91, and then let you know.

Reply:
yes, a wrong value of `id` on line 25 would cause another retransmission (and the test failure on a patched kernel):

```
# git diff
diff --git a/gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt b/gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt
index 754cf4c..226ce2f 100644
--- a/gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt
+++ b/gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt
@@ -24,7 +24,7 @@
 +0       <   .  1:1(0)        ack 1     win 257  <add_address address_id=90 addr[saddr0] addr_echo>
 +1.0     >   .  1:1(0)        ack 1              <add_address address_id=1  addr[saddr0] hmac=auto,                 dss dack4=1 ssn=1 dll=0 nocs>
 // send echo with wrong id, expect  no retry
-+0       <   .  1:1(0)        ack 1     win 257  <add_address address_id=1 addr[saddr0] addr_echo>
++0       <   .  1:1(0)        ack 1     win 257  <add_address address_id=91 addr[saddr0] addr_echo>
 // read and ack 1 data segment
 +1.1     <  P.  1:1001(1000)  ack 1     win 450  <nop, nop,                                                         dss dack8=1    dsn8=1 ssn=1 dll=1000 nocs>
 +0       >   .  1:1(0)        ack 1001                                                                             <dss dack8=1001        ssn=1 dll=0 nocs>
# ./packetdrill/run_all.py -v -l mptcp/add_addr/add_addr_retry_v4.pkt 
FAIL [/root/packetdrill/gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt (ipv4)]
stdout: 
net.mptcp.add_addr_timeout = 1
stderr: 
add_addr_retry_v4.pkt:30: error handling packet: live packet field ipv4_total_length: expected: 52 (0x34) vs actual: 64 (0x40)
script packet:  3.177133 . 1:1(0) ack 1001 <dss dack8 13263177308786789773 flags: Aa>
actual packet:  3.100095 . 1:1(0) ack 1 win 256 <add_address address_id: 1 ipv4: 192.168.0.3 hmac: 9085218341433227832,dss dack4 3007449509 flags: A>
FAIL [/root/packetdrill/gtests/net/mptcp/add_addr/add_addr_retry_v4.pkt (ipv4-mapped-v6)]
stdout: 
net.mptcp.add_addr_timeout = 1
stderr: 
add_addr_retry_v4.pkt:30: error handling packet: live packet field ipv4_total_length: expected: 52 (0x34) vs actual: 64 (0x40)
script packet:  3.230830 . 1:1(0) ack 1001 <dss dack8 13263177308786789773 flags: Aa>
actual packet:  3.154213 . 1:1(0) ack 1 win 256 <add_address address_id: 1 ipv4: 192.168.0.3 hmac: 8524091128375274721,dss dack4 3007449509 flags: A>
Ran    2 tests:    0 passing,    2 failing,    0 timed out (5.02 sec): mptcp/add_addr/add_addr_retry_v4.pkt
# 
```

Reply:
> > with "right" ID? (same in the `_v6` version)
> 
> correct,
> 
done